keel-core/src/main/kotlin/com/netflix/spinnaker/keel/actuation/EnvironmentConstraintRunner.kt

```diff
@@ -133,18 +134,18 @@ class EnvironmentConstraintRunner(
    */
   private fun handleOlderPendingVersions(
     envContext: EnvironmentContext,
-    pendingVersionsToCheck: MutableSet<String>
+    pendingVersionsToCheck: MutableList<PublishedArtifact>
   ) {
     log.debug("pendingVersionsToCheck: [$pendingVersionsToCheck] of artifact ${envContext.artifact.name} for environment ${envContext.environment.name} ")
     pendingVersionsToCheck
-      .sortedWith(envContext.artifact.versioningStrategy.comparator.reversed()) // oldest first
-      .forEach { version ->
+      .reversed() // oldest first
```

Comment:
These versions should already be sorted when we get here.

Reply:
Is there a test that verifies this?

Reply:
Good point. No, I'll try to add some.

Reply:
Added in https://github.com/spinnaker/keel/pull/1652/commits/c653644f2d8e68d445fd47f79977c902308d77c3